kong/tracing/propagation.lua

```diff
@@ -38,6 +41,18 @@ local function left_pad_zero(str, count)
   return ('0'):rep(count-#str) .. str
 end
 
+
+local function to_w3c_trace_id(trace_id)
+  if #trace_id < W3C_TRACEID_LEN then
+    return ('\0'):rep(W3C_TRACEID_LEN - #trace_id) .. trace_id
+  elseif #trace_id > W3C_TRACEID_LEN then
+    return trace_id:sub(-W3C_TRACEID_LEN)
+  end
+
+  return trace_id
+end
```

Comment:
It would better if we have a single "to_provider_trace_id(trace_id,provider_type)" and unit test the heck out of it instead of ensuring that we call the right function in the right if branch below.
Not a blocker for this PR.

Reply:
Agree. But let's wait until another larger round of refactoring to fix long-term issues.